plugins/inputs/dcos_containers/dcos_containers.go

```diff
@@ -0,0 +1,310 @@
+	warnIfNotSet(setIfNotNil(net.fields, "tx_packets", rs.GetNetTxPackets))
+	warnIfNotSet(setIfNotNil(net.fields, "tx_bytes", rs.GetNetTxBytes))
+	warnIfNotSet(setIfNotNil(net.fields, "tx_errors", rs.GetNetTxErrors))
+	warnIfNotSet(setIfNotNil(net.fields, "tx_dropped", rs.GetNetTxDropped))
+	warnIfNotSet(setIfNotNil(net.fields, "tcp_rtt_microsecs_p50", rs.GetNetTCPRttMicrosecsP50))
+	warnIfNotSet(setIfNotNil(net.fields, "tcp_rtt_microsecs_p90", rs.GetNetTCPRttMicrosecsP90))
+	warnIfNotSet(setIfNotNil(net.fields, "tcp_rtt_microsecs_p95", rs.GetNetTCPRttMicrosecsP95))
+	warnIfNotSet(setIfNotNil(net.fields, "tcp_rtt_microsecs_p99", rs.GetNetTCPRttMicrosecsP99))
+	warnIfNotSet(setIfNotNil(net.fields, "tcp_active_connections", rs.GetNetTCPActiveConnections))
+	warnIfNotSet(setIfNotNil(net.fields, "tcp_time_wait_connections", rs.GetNetTCPTimeWaitConnections))
+	// TODO (philipnrmn) *rs.NetTrafficControlStatistics  for net traffic control statistics
+	// TODO (philipnrmn) *rs.NetSNMPStatistics for net snmp statistics
+
+	return results
+}
+
+// cBlkioMeasurement flattens the deeply nested blkio statistics struct into
+// a set of measurements, tagged by device ID and blkio policy
+func cBlkioMeasurements(bs mesos.CgroupInfo_Blkio_Statistics) []measurement {
+	var results []measurement
+
+	for _, cfq := range bs.GetCFQ() {
+		blkio := newMeasurement("blkio")
+		blkio.tags["policy"] = "cfq"
+		if dev := cfq.GetDevice(); dev != nil {
+			blkio.tags["device"] = fmt.Sprintf("%d.%d", dev.GetMajorNumber(), dev.GetMinorNumber())
+		} else {
+			blkio.tags["device"] = "default"
+		}
+		warnIfNotSet(setIfNotNil(blkio.fields, "io_serviced", blkioTotalGetter(cfq.GetIOServiced)))
+		warnIfNotSet(setIfNotNil(blkio.fields, "io_service_bytes", blkioTotalGetter(cfq.GetIOServiceBytes)))
+		warnIfNotSet(setIfNotNil(blkio.fields, "io_service_time", blkioTotalGetter(cfq.GetIOServiceTime)))
+		warnIfNotSet(setIfNotNil(blkio.fields, "io_wait_time", blkioTotalGetter(cfq.GetIOWaitTime)))
+		warnIfNotSet(setIfNotNil(blkio.fields, "io_merged", blkioTotalGetter(cfq.GetIOMerged)))
+		warnIfNotSet(setIfNotNil(blkio.fields, "io_queued", blkioTotalGetter(cfq.GetIOQueued)))
+
+		results = append(results, blkio)
+	}
+
+	return results
+}
+
+// blkioTotalGetter is a convenience method allowing us to unpick the nested
+// blkio_value object. It returns a method which when invoked, returns the
+// value of the total of the field returned by its parameter function
+func blkioTotalGetter(f func() []mesos.CgroupInfo_Blkio_Value) func() uint64 {
+	return func() uint64 {
+		for _, v := range f() {
+			// TODO (philipnrmn) consider returning all operations, not just total
+			if v.GetOp() == mesos.CgroupInfo_Blkio_TOTAL {
+				return v.GetValue()
+			}
+		}
+		return 0
+	}
+}
+
+// cTags extracts relevant metadata from a Container object as a map of tags
+func cTags(c agent.Response_GetContainers_Container) map[string]string {
+	return map[string]string{"container_id": c.ContainerID.Value}
```

Comment:
Should this be `dcos_container_id`? I've been assuming that the tags we add for DC/OS-specific values (e.g. cluster ID, node role) would be prefixed with `dcos_`.

Reply:
Tricky one. I have been using `container_id` since container IDs are mesos-level rather than DC/OS-level concepts. But then this whole plugin is `dcos_containers` so maybe that's wrong. Do you have strong feelings either way?

Reply:
My only concern is whether the metrics API output can identify the metrics that come from this input, so that it can pass them along to the HTTP producer.  `container_id` seems a like a generic enough tag that looking just for that risks creating false positives, though realistically it probably won't cause any problems. I'll see if I can find other hints to use, and fall back to just looking for a `container_id` tag if I can't.